packages/fast-jest-snapshots-react/src/index.ts

```diff
@@ -1,6 +1,18 @@
 import * as React from "react";
 import * as renderer from "react-test-renderer";
 
+/**
+ * A set of snapshot test cases
+ */
+export type ISnapshotTestCases<T> = Array<ISnapshotTestCase<T>>;
+
+/**
+ * An prop object to supply for snapshots. Optionally, a
+ * an array can be provided where the first index is a string representing
+ * a snapshot description, with the second index being the prop data
+ */
+export type ISnapshotTestCase<T> = T | [string, T];

```

Comment:
We can't just use an object because there would be no safe way to differentiate between a prop object and a test-case object, because any structure we use could be valid react props.

What is the benefit of using a class from your perspective? My initial thought is it's overly-ceremonious for little benefit. All we really need is a different data structure than an object and an array fills that need. It's also a native language feature, so we wouldn't need any additional imports and we wouldn't need to instantiate a new class every test, which IMO results in a less cluttered test suite:

```
[
  [
    "should do a thing",
    {
      message: "hello world"
    }
  ],
  [
    "should do another thing",
    {
      message: "hello world - revisited"
    }
  ]
]

// vs
[
  new NamedSnapshotTestCase(
    "Should do a thing",
    {
      message: "hello world"
    }
  ),
  new NamedSnapshotTestCase(
    "Should do another thing",
    {
      message: "hello world - revisitied"
    }
  )
]

Reply:
I don't like the class thing either because it's too heavy handed. The only reason I proposed is because it would let you differentiate between normal props and a test description object (i.e., via X instanceof TestCase), etc. It would also make future extensions easier (than just adding each thing as a new element in the array).

Since you've defined it as a tuple of [string, T], this should presumably give some better intellisense in VSCode, for example (of course well named props would be better, but I think this is a good tradeoff).